packages/splits-sdk-react/src/hooks/data.ts

```diff
@@ -0,0 +1,850 @@
+
+  const [swapperMetadata, setSwapperMetadata] = useState<Swapper | undefined>()
+  const [isLoading, setIsLoading] = useState(!!swapperAddress)
+  const [status, setStatus] = useState<DataLoadStatus | undefined>(
+    swapperAddress ? 'loading' : undefined,
+  )
+  const [error, setError] = useState<RequestError>()
+
+  useEffect(() => {
+    let isActive = true
+
+    const fetchMetadata = async () => {
+      try {
+        const swapper = await splitsClient.getSwapperMetadata({
+          chainId,
+          swapperAddress,
+        })
+        if (!isActive) return
+        setSwapperMetadata(swapper)
+        setStatus('success')
+      } catch (e) {
+        if (isActive) {
+          setStatus('error')
+          setError(e)
+        }
+      } finally {
+        if (isActive) setIsLoading(false)
+      }
+    }
+
+    setError(undefined)
+    if (swapperAddress) {
+      setIsLoading(true)
+      setStatus('loading')
+      fetchMetadata()
+    } else {
+      setStatus(undefined)
+      setIsLoading(false)
+      setSwapperMetadata(undefined)
+    }
+
+    return () => {
+      isActive = false
+    }
+  }, [splitsClient, swapperAddress])
+
+  return {
+    isLoading,
+    swapperMetadata,
+    status,
+    error,
+  }
+}
+
+const DEFAULT_OPTIONS = {
+  contractIds: undefined,
+}
+
+export const useUserEarnings = (
+  chainId: number,
+  userAddress: string,
+): {
+  isLoading: boolean
+  userEarnings?: UserEarnings
+  status?: DataLoadStatus
+  error?: RequestError
+} => {
+  const context = useContext(SplitsContext)
+  const splitsClient = getSplitsClient(context).dataClient
+
+  if (!splitsClient) throw new Error('Missing api key for data client')
+
+  const [userEarnings, setUserEarnings] = useState<UserEarnings | undefined>()
+  const [isLoading, setIsLoading] = useState(!!userAddress)
+  const [status, setStatus] = useState<DataLoadStatus | undefined>(
+    userAddress ? 'loading' : undefined,
+  )
+  const [error, setError] = useState<RequestError>()
+
+  useEffect(() => {
+    let isActive = true
+
+    const fetchEarnings = async () => {
+      try {
+        const earnings = await splitsClient.getUserEarnings({
+          chainId,
+          userAddress,
+        })
+        if (!isActive) return
+        setUserEarnings(earnings)
+        setStatus('success')
+      } catch (e) {
+        if (isActive) {
+          setStatus('error')
+          setError(e)
+        }
+      } finally {
+        if (isActive) setIsLoading(false)
+      }
+    }
+
+    setError(undefined)
+    if (userAddress) {
+      setIsLoading(true)
+      setStatus('loading')
+      fetchEarnings()
+    } else {
+      setStatus(undefined)
+      setIsLoading(false)
+      setUserEarnings(undefined)
+    }
+
+    return () => {
+      isActive = false
+    }
+  }, [splitsClient, userAddress])
+
+  return {
+    isLoading,
+    userEarnings,
+    status,
+    error,
+  }
+}
+
+export const useFormattedUserEarnings = (
```

Comment:
nit: i think now that our gql api has token data (symbol and decimals), we should just make the default user formatted earnings and get rid of this second function. does that sound right? only made it a separate function because it required extra calls that i figured people might not care about. but if all the data is in the subgraph now don't see a reason to never not return the formatted version.

don't think you need to update that in this pr. i think i definitely want to make that change though before we publish 4.0.0 (doesn't need to hold up any alpha or beta's though)

Reply:
https://linear.app/splits/issue/PE-1953/sdk-have-all-earnings-be-formatted-earnings